KMS v2alpha1 (#35046)
kubernetes/website: content/en/docs/reference/config-api/apiserver-encryption.v1.md

<thead><tr><th width="30%">Field</th><th>Description</th></tr></thead>
<tbody>
    

> <tr><td><code>apiversion</code><br/>

Reviewer: ```suggestion
<tr><td><code>apiVersion</code><br/>
```

> <tr><td><code>apiVersion</code><br/>

Reviewer 2: This file is auto-generated. Please don't modify manually.

Author: Thanks! I will remove it. Would the change get automatically added at some point? What triggers the auto-gen?

Reviewer 2: New version is generated when a new k8s release is out, maybe a few days post the release.

<code>string</code>
</td>
<td>

>    <p>apiversion is the API Version for KMS provider (Allowed values: v2, empty or anything else will default fallback to v1)</p>

Reviewer: ```suggestion
   <p>apiVersion is the API Version for KMS provider (Allowed values: v2, v1 or empty. Any other value will result in an error.)</p>
```

nit: As part of the implementation we decided to default empty to `v1` and in validation we check if it's either `v1` or `v2`. Any other API version will result in an error.

ref: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/apis/config/validation/validation.go#L225-L232
